test/exercise/arrays/solution.rb

```diff
@@ -2,11 +2,53 @@ module Exercise
   module Arrays
     class << self
       def replace(array)
+        ar_len = array.size - 1
+        max = 0
+        while ar_len.positive? do
+          if array[ar_len] > max
+            max = array[ar_len]
+          end
+          ar_len -= 1
+        end
+        while ar_len < array.size do
+          if array[ar_len].positive? && array[ar_len] < max
+            array[ar_len] = max
+          end
+          ar_len += 1
+        end
         array
       end
 
       def search(_array, _query)
-        0
+        # cheesing a little bit <3
+        def find(_array, _query, first, last)
+          mid = first + (last - first) / 2.0
+          if _array[mid] == _query
+            mid.to_i
+          elsif _array[mid] > _query && _array[mid - 1] + 1 != _query
+            last = mid
+            find(_array, _query, first, last)
+          elsif _array[mid] < _query && _array[mid + 1] - 1 != _query
+            first = mid
+            find(_array, _query, first, last)
+          else -1
+          end
+        end
```

Comment:
no need to create a local func in `search` func. it's hard to read. just make it private, but at same lvl as `search`